File: src/dotnet/APIView/APIView/Model/CodeDiagnostic.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

using Newtonsoft.Json;

namespace APIView
{
    public class CodeDiagnostic
    {
        public CodeDiagnostic()
        {
        }

        public CodeDiagnostic(string diagnosticId, string targetId, string text, string helpLinkUri, CodeDiagnosticLevel level = CodeDiagnosticLevel.Error)
        {
            DiagnosticId = diagnosticId;
            Text = text;
            TargetId = targetId;
            HelpLinkUri = helpLinkUri;
            Level = level;
        }
        [JsonProperty("di")]
        public string DiagnosticId { get; set; }
        [JsonProperty("t")]
        public string Text { get; set; }
        [JsonProperty("hlu")]
        public string HelpLinkUri { get; set; }
        [JsonProperty("ti")]
        public string TargetId { get; set; }
        [JsonProperty("l")]
        public CodeDiagnosticLevel Level { get; set; }
    }
}

File: src/dotnet/APIView/ClientSPA/src/app/_models/codeDiagnostic.ts
export class CodeDiagnostic {
    diagnosticId: string = '';
    text: string = '';
    helpLinkUri: string = '';
    targetId: string = '';
    level: string = '';

    constructor(
        diagnosticId: string = '',
        text: string = '',
        helpLinkUri: string = '',
        targetId: string = '',
        level: string = ''
    ) {
        this.diagnosticId = diagnosticId;
        this.text = text;
        this.helpLinkUri = helpLinkUri;
        this.targetId = targetId;
        this.level = level;
    }
}